compiler/luci/service/src/CircleShapeSignatureInferenceHelper.cpp

```diff
@@ -66,15 +75,18 @@ ShapeSignature reduced_signature(const loco::Node *node, const loco::Node *indic
     else
     {
       // There is no way to inference for this case.
-
-      // This assert is just to find this case occurs at debug version.
-      assert(false && "Cannot infer reduced_signature");
+      // Do nothing to return empty signature.
+      INFO(l) << "[CircleShapeSignatureInferenceHelper] " << circle_node->name() << std::endl;
+      INFO(l) << " reduced_signature : cannot infer because of non-constant node" << std::endl;
     }
 
     return output_signature;
   }
 
-  // If shape signature is empty, output is static.
+  // If input rank is 0, it means that one of following case is occurred.
+  // - Input is scalar : result is always scalar
+  // - Input shape signature is not inferenced : cannot infer output shape signauture
+  // Therefore, when input signature rank is 0, always return empty signature.
```

Comment:
I've got two question about this lines.
1. It's not related with this PR. I think below condition can be moved to above line 65. Because even if either `reduction_indices` is `nullptr` or not, there is no difference for `output_signiture`'s being scalar.
2. Could you let me know about the second one more detail?; `Input shape signature is not inferenced` Because of below lines, this circumstance wouldn't happen I think.
```cxx
else // maybe the case of `input_signature.rank()==0`
  {
    input_signature.rank(circle_node->rank());
    for (uint32_t i = 0; i < circle_node->rank(); ++i)
      input_signature.dim(i) = circle_node->dim(i).value();
  }
```

Reply:
1. Before I created this PR, below condition should be there but for now, I think you are right. I will move it.

1. It can happens. There are some cases in TensorFlow and TFLite that shape cannot be inferred.
For example, let's think about following codes.
```python
inp = tf.compat.v1.placeholder(tf.float32, shape=(2,3,4))
idx = tf.compat.v1.placeholder(tf.int32, shape=(2))
r_mean1 = tf.compat.v1.math.reduce_mean(inp, idx, False)
r_mane2 = tf.compat.v1.math.reduce_mean(r_mean1, [0], True)
```
Because shape of `r_mean1` cannot be inferred, both shape and shape signature of `r_mean1` are `[]`.
In this case, `input_signature` is `[]`, which rank is 0.